ait/core/val.py

```diff
@@ -360,30 +359,118 @@ def pretty(self, start, end, e, messages=None):
 
-        The schema file should validate against JSON Schema Draft 4
+        - The schema file should validate against JSON Schema Draft 4
         http://json-schema.org/latest/json-schema-core.html
 
-        The YAML file should validate against the schema file given
+        - The YAML file should validate against the schema file given
         """
         self._ymlfile = ymlfile
         self._schemafile = schemafile
-
+        self._ymlproc = None
         # Get the error handler ready, just in case
         self.ehandler = ErrorHandler(ymlfile=self._ymlfile, schemafile=self._schemafile)
+        # Declare variables used for validating nested yaml files
+        self.validate_list = []
+        self.yml_dir = f"{ymlfile.rsplit('/', 1)[0]}/"
+        self.yml_files_to_validate = self.get_included_files(ymlfile)
+
+    def get_included_files(self, yml_file):
+            """
+            Make a list of  yaml files to validate against the schema.  The first member
+            of the list will include the full path to the yaml file.
+            The next element will the module yml file (e.g. cmd.yml),
+            followed by the names of any included files.
+            The assumption is made that all the included yml files are
+            found in the same directory as the module yml file.
+
+            Parameters
+            ----------
+            yml_file : str
+                The module yaml file to validate (cmd.yml or tlm.yml)
+
+            Returns
+            -------
+            self.validate_list: list
+                A set of all files that are to be validated.
+
+            """
+            # The first yaml file to validate will include a full path
+            # Any included yaml files will have the path added.
+            if '/' in yml_file:
+                self.validate_list.append(yml_file)
+            else:
+                yml_file = f'{self.yml_dir}/{yml_file}'
+
+            try:
+                with open(yml_file, "r") as yml_fh:
+                    for line in yml_fh:
+                        if not line.strip().startswith("#") and "!include" in line:
+                            included_file_name = line.split('!include ')[-1].strip()
+                            self.validate_list.append(f'{self.yml_dir}/{included_file_name}')
+                            # Look for includes within included file
+                            self.get_included_files(included_file_name)
+                    # Check and flag multiple includes
+                    if len(self.validate_list) != len(set(self.validate_list)):
+                        log.info(f'WARNING: Validate: Duplicate config files in the '
+                                 f'include tree. {self.validate_list}')
+                    return set(self.validate_list)
+            except RecursionError as e:
+                print(f'ERROR: {e}: Infinite loop: check that yaml config files are not looping '
```

Comment:
Switch to logger call

Reply:
Done, also made the same change in util.py, where there is similar code.